.github/workflows/integration_test.yaml

```
@@ -81,7 +68,13 @@ jobs:
           " >> /var/snap/microk8s/current/args/containerd-template.toml'
           sudo su -c 'systemctl restart snap.microk8s.daemon-containerd.service && microk8s status --wait-ready'
       - name: Run integration tests
-        run: tox -e integration -- --model testing --indico-image ${{ env.REGISTRY }}/${{ env.OWNER }}/${{ env.INDICO_IMAGE_NAME }}:${{ github.run_id }} --indico-nginx-image ${{ env.REGISTRY }}/${{ env.OWNER }}/${{ env.INDICO_NGINX_IMAGE_NAME }}:${{ github.run_id }}
+        run: |
+          image_names=$(echo '${{ needs.get-images.outputs.images }}' | jq -cr '.[]')
```

Could it be worth formatting the output such that further processing isn't required? It would simplify this a bit

---

Not sure it's possible github actions enforce the output format as json while bash can't iterate on json array directly
But we can directly iterate in the for loop from the output of JQ (prehaps less readable)
```suggestion
          for image_name in $(echo '${{ needs.get-images.outputs.images }}' | jq -cr '.[]'); do
```

---

The best I could do is:
```
args=$(echo '${{ needs.get-images.outputs.images }}' | jq -cr '.[]' | \
            xargs -I{} echo "--${image_name}-image ${{ env.REGISTRY }}/${{ env.OWNER }}/${image_name}:${{ github.run_id }}" | \
            paste -sd' ' -
```
Any suggestions?

---

I think the for loop is more readable than the xargs / paste

---

Agreed